File: Domain/Founder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Domain
{
    public class Founder
    {
        public Founder()
        {
            this.rFirstName = "";
            this.rLastName = "";
            this.rSub = "";
            this.rTitle = "";
            this.rContactMail = "";
            this.rEnable = true;
            this.rCreate = DateTime.Now;
            this.rExpire = DateTime.Now.AddMonths(1);
        }

        public Guid Id { get; set; }
        public string rFirstName { get; set; }
        public string rLastName { get; set; }
        public string rSub { get; set; }
        public string rTitle { get; set; }
        public string rContactMail { get; set; }
        public bool rEnable { get; set; }
        public DateTime rCreate { get; set; }
        public DateTime rExpire { get; set; }
        public virtual AppUser AppUser { get; set; }
        public string AppUserId { get; set; }

    }
}

File: client-app/src/app/models/founder.ts
export interface IFounder {
  id: string;
  rSub: string;
  rTitle: string;
  rContactMail: string;
  rEnable: boolean;
  rCreate: Date | string;
  rExpire: Date | string;
  rCreateJalali: string;
  rExpireJalali: string;
}

export class Founder implements IFounder {
  id: string = '';
  rSub: string = '';
  rTitle: string = '';
  rContactMail: string = '';
  rEnable: boolean = true;
  rCreate: string | Date = '';
  rExpire: string | Date = '';
  rCreateJalali: string = '';
  rExpireJalali: string = '';

  constructor(init?: IFounder) {
    Object.assign(this, init);
  }
}
